src/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreCLR.cs

```diff
@@ -228,6 +228,36 @@ public static void GetAvailableThreads(out int workerThreads, out int completion
             GetAvailableThreadsNative(out workerThreads, out completionPortThreads);
         }
 
+        /// <summary>
+        /// Gets the number of thread pool threads that currently exist.
+        /// </summary>
+        /// <remarks>
+        /// For a thread pool implementation that may have different types of threads, the count includes all types.
+        /// </remarks>
+        public static extern int ThreadCount
+        {
+            [MethodImpl(MethodImplOptions.InternalCall)]
+            get;
+        }
+
+        /// <summary>
+        /// Gets the number of work items that have been processed so far.
+        /// </summary>
+        /// <remarks>
+        /// For a thread pool implementation that may have different types of work items, the count includes all types.
+        /// </remarks>
+        public static extern long CompletedWorkItemCount
+        {
+            [MethodImpl(MethodImplOptions.InternalCall)]
+            get;
```

Comment:
Interesting; I wasn't expecting to see this implemented in native.  Why is that necessary?  I'm also missing where this is factoring in work items dequeued and executed from the managed queues.  Where does that cause the count to be incremented?

Reply:
Timer, wait, and IO completion callbacks are initiated from the native side and there is already thread-local-ish code there to track completion counts for hill climbing, so I just built on top of that.

Reply:
For work items in the managed queues this call increments the count:
https://github.com/dotnet/coreclr/blob/a006c44d06d56829a230bbbce17b74b21a0b8842/src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs#L647-L652

Here:
https://github.com/dotnet/coreclr/blob/a006c44d06d56829a230bbbce17b74b21a0b8842/src/vm/win32threadpool.h#L842-L847